File: ios/HybridTextRecognizerFactory.swift
//
//  HybridTextRecognizerFactory.swift
//  VisionCameraTextScanner
//
//  Created by Alex Korn on 21.04.26.
//

import NitroModules
import VisionCamera

class HybridTextRecognizerFactory: HybridTextRecognizerFactorySpec {
  func createTextRecognizer(options: TextRecognizerOptions) throws -> any HybridTextRecognizerSpec {
    return HybridTextRecognizer(options: options)
  }

  func createTextRecognizerOutput(options: TextRecognizerOutputOptions) throws
    -> any HybridCameraOutputSpec
  {
    return HybridTextRecognizerOutput(options: options)
  }
}

File: android/src/main/java/com/margelo/nitro/camera/textrecognizer/HybridTextRecognizerFactory.kt
package com.margelo.nitro.camera.textrecognizer

import com.margelo.nitro.camera.HybridCameraOutputSpec

class HybridTextRecognizerFactory : HybridTextRecognizerFactorySpec() {
  override fun createTextRecognizer(options: TextRecognizerOptions): HybridTextRecognizerSpec {
    return HybridTextRecognizer(options)
  }

  override fun createTextRecognizerOutput(
    options: TextRecognizerOutputOptions
  ): HybridCameraOutputSpec {
    return HybridTextRecognizerOutput(options)
  }
}